llvm/lib/Transforms/InstCombine/InstCombineSelect.cpp

```diff
@@ -996,6 +989,23 @@ static Value *canonicalizeSaturatedAdd(ICmpInst *Cmp, Value *TVal, Value *FVal,
   if (!match(TVal, m_AllOnes()))
     return nullptr;
 
+  if (Pred == ICmpInst::ICMP_UGE &&
+      match(FVal, m_Add(m_Value(X), m_APInt(C))) && X == Cmp0 &&
+      match(TVal, m_AllOnes()) && match(Cmp1, m_APInt(CmpC)) && *CmpC == ~*C) {
```

Comment:
```suggestion
      match(Cmp1, m_APInt(CmpC)) && *CmpC == ~*C) {
```
Already checked above.